test/namespace.test.js

```diff
@@ -0,0 +1,57 @@
+var API = require('../src/api')
+var assert = require('assert')
+
+describe('Namespace', function() {
+
+  it ('can namespace a route', function() {
+    var api = new API()
+    var foo = api.namespace('foo')
+    var bar = foo.namespace('bar')
+
+    assert.deepEqual(bar.segments, [ 'foo', 'bar' ])
+  })
+
+  it ('does not clobber namespaces', function() {
+    var api = new API()
+
+    api.route({
+      users: {
+        read: {}
+      }
+    })
+
+    var users = api.namespace('users')
+
+    assert('read' in api.users)
+  })
+
+  it ('toString reflects namespace segments', function() {
+    var api = new API()
+    var foo = api.namespace('foo')
+    var bar = foo.namespace('bar')
+
+    assert.equal(bar.toString(), '/foo/{foo_id}/bar')
+  })
+
+  it ('toString singularizes parameters segments', function() {
+    var api = new API()
+    var users = api.namespace('users')
+    var posts = users.namespace('posts')
+
+    assert.equal(posts.toString(), '/users/{user_id}/posts')
```

Comment:
Good question. I don't think so. You absolutely need an `id` parameter for `update` and `delete`.